bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapWeatherAndForecastHandler.java

```diff
@@ -298,6 +298,9 @@ private void updateCurrentChannel(ChannelUID channelUID) {
                     state = getQuantityTypeState(weatherData.getSnow() == null ? 0 : weatherData.getSnow().getVolume(),
                             MILLI(METRE));
                     break;
+                case CHANNEL_VISIBILITY:
+                    state = getQuantityTypeState(weatherData.getVisibility() * 0.001, KILO(METRE));
```

Comment:
Use the power of the `QuantityType` to do the math - e.g.:

```
new QuantityType<>(weatherData.getVisibility(), METRE).toUnit(KILO(METRE));
```

But the question is why do you want to modify the source value at all? Let the user decide which unit he or she likes. You maybe can add an example to the README which converts the unit to "km" like I did for the `wind-speed` channel.

Reply:
Please give a document link for getQuantityTypeState, I want to learn how to use it and how to set unit in configuration. 

And I still have a question about that:  
1, how this visibility store in mysql database? meter or kilometer ? I have other rules/scripts/webpages, I hope they are in same units in datebase. 

2, What will happen if I have a chart to show a group of visibilities, other values are from weatherungerground and in unit kilometer?

Reply:
Here are two links: [our documentation](https://www.openhab.org/docs/concepts/units-of-measurement.html) and a [blog post](https://www.eclipse.org/smarthome/blog/2018/02/22/units-of-measurement.html) published after the feature has been introduced.

1. Currently the persistent is not aware of UoM and quantities. Meaning it just stores the decimal value of the state. If you will use "mm" on the one hand it will store the value in "mm", on the other hand if you use "km" the value will be stored in "km". There are some issues around describing problems related to that (see https://github.com/eclipse/smarthome/issues/5675 or  https://github.com/eclipse/smarthome/issues/5749).

2. See above.

Reply:
I want to ask if single UTF-8 character like "㎞" is support, where can I check the code or add features. 

I have some UTF-8 character of unit in my items config file:
----------------------------------------------------------------
㎍   (ug)
㎜    (mm)
㎝   (cm)
㎞   (km)
㎡   (m2)
㎥   (m3)
㏙   (PPW)
℃   (°C),
㍱   (hPa)
㏓   (Lx)

Reply:
The UoM itself is an external library (https://github.com/unitsofmeasurement/uom-se). Basic features are implemented in openHAB Core (https://github.com/openhab/openhab-core). Currently we only handle the special char for "℃" - just search the core repository for "[°C](https://github.com/openhab/openhab-core/search?p=2&q=%C2%B0C&unscoped_q=%C2%B0C)" or "[QuantityType](https://github.com/openhab/openhab-core/search?q=QuantityType&unscoped_q=QuantityType)".

Reply:
Can we using kilometer as default unit. 10㎞ looks better than 10km or 10000m in my basic UI, I want use unit meter after "㎞" is supported.

Reply:
As I mentioned above: To support the unicode characters we need a feature request in openHAB Core.